### PR TITLE
chore: the chorest chore of all: reanimated upgrade (third try, metaverse edition)

### PR DESCRIPTION
**Pull request by pvinis**
This PR resolves []

### QA Test Case(s)

screens with animations to QA. make sure things work as they should:
- onboarding welcome
- search, input and cancel
- phone number edit in profile
- animatable header. screens with filters, like a gene page
- MyCollectionSearchBar
- MyCollection
- MyCollectionArtworks

### PR Checklist

- [ ] I tested my changes on **iOS** / **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: android/app/src/main/java/net/artsy/app/MainActivity.java [hunk 1]
```diff
@@ -34,16 +33,6 @@ protected String getMainComponentName() {
     return "eigen";
   }
 
-  @Override
-  protected ReactActivityDelegate createReactActivityDelegate() {
```
Comment (pvinis) [inline android/app/src/main/java/net/artsy/app/MainActivity.java:38]: this comes from the upgraded gesture-handler dep

File: babel.config.js [hunk 2]
```diff
@@ -4,44 +4,28 @@ module.exports = (api) => {
...
-    presets: [
-      [
-        "module:metro-react-native-babel-preset",
-        {
-          unstable_disableES6Transforms: true,
-          useTransformReactJSXExperimental: true, // this is so `import React from "react"` is not needed.
-        },
-      ],
-    ],
+    // plugins run first
```
Comment (pvinis) [inline babel.config.js:7]: better babelrc to support reanimated

File: src/app/Components/ArtworkFilter/Filters/WaysToBuyOptions.tests.tsx [hunk 3]
```diff
@@ -43,61 +41,6 @@ describe("Ways to Buy Options Screen", () => {
     expect(getByText("Inquire")).toBeTruthy()
   })
 
-  it("does not display the default text when no filter selected on the filter modal screen", () => {
```
Comment (pvinis) [inline src/app/Components/ArtworkFilter/Filters/WaysToBuyOptions.tests.tsx:46]: bring back and .skip

**Comment by pvinis** [comment]
- list of animation screens for QA
- list of useAnimatedValue for a spike to v1->v2 migration

**Comment by pvinis** [comment]
just adding you two for fun :)

File: EXAMPLES.md [hunk 4]
```diff
@@ -102,3 +102,15 @@ One good example is in [OnboardingLogin.tsx](src/app/Scenes/Onboarding/Onboardin
...
+
+We use `react-native-reanimated` for that.
+Here are some examples.
+
+Links:
+
+- [OnboardingWelcome.tsx](src/app/Scenes/Onboarding/OnboardingWelcome.tsx)
+- [Search.tsx](src/app/Scenes/Search/Search.tsx)
+- [AnimatableHeaderLargeTitle.tsx](src/app/Components/AnimatableHeader/AnimatableHeaderLargeTitle.tsx)
+- [AnimatableHeader.tsx](src/app/Components/AnimatableHeader/AnimatableHeader.tsx)
```
Comment (MounirDhahri) [inline EXAMPLES.md:116]: **issue(non-blocker)**: I would actually recommend using the Animated API and then as a second option to go with reanimated

File: src/app/Components/StickyTabPage/StickyTabPageFlatList.tsx [hunk 5]
```diff
@@ -137,6 +136,7 @@ export const StickyTabPageFlatList: React.FC<StickyTabFlatListProps> = (props) =
             }}
           />
         }
+        // @ts-expect-error
```
Comment (MounirDhahri) [inline src/app/Components/StickyTabPage/StickyTabPageFlatList.tsx:139]: 👀

File: src/app/Scenes/MyCollection/Components/MyCollectionSearchBar.tsx [hunk 6]
```diff
@@ -69,13 +66,10 @@ export const MyCollectionSearchBar: React.FC<MyCollectionSearchBarProps> = ({
...
       Animated.call(
         [staticHeaderHeight, hasRunFocusedAnimation],
-        ([staticHeaderHeightValue, hasFinishedAnimationLoop]) => {
+        ([, hasFinishedAnimationLoop]) => {
           if (hasFinishedAnimationLoop) {
             return
           }
-          innerFlatListRef?.current
-            ?.getNode()
-            .scrollToOffset({ offset: Number(staticHeaderHeightValue), animated: true })
```
Comment (MounirDhahri) [inline src/app/Scenes/MyCollection/Components/MyCollectionSearchBar.tsx:78]: **thought:** does this break any animation in my collection search bar?

**Review by MounirDhahri — APPROVED** [review]

File: src/app/Scenes/MyCollection/Components/MyCollectionSearchBar.tsx (hunk 6 above)
Comment (pvinis) [reply src/app/Scenes/MyCollection/Components/MyCollectionSearchBar.tsx:78]: to be tested in the beta and QA. I am guessing it didn't. if it did, we will fix it :)

Comment (MounirDhahri) [reply src/app/Scenes/MyCollection/Components/MyCollectionSearchBar.tsx:78]: oki - cool. anyway this is not a blocker for us in CX 🙏

File: android/app/src/main/java/net/artsy/app/MainActivity.java [hunk 7]
```diff
@@ -34,16 +33,6 @@ protected String getMainComponentName() {
...
-    return new ReactActivityDelegate(this, getMainComponentName()) {
-      @Override
-      protected ReactRootView createRootView() {
-        return new RNGestureHandlerEnabledRootView(MainActivity.this);
-      }
-    };
-  }
-
   private boolean isTablet() {
```
Comment (brainbicycle) [inline android/app/src/main/java/net/artsy/app/MainActivity.java:36]: 👋

File: ios/Podfile.lock [hunk 8]
```diff
@@ -1239,6 +1264,6 @@ SPEC CHECKSUMS:
   Yoga: e7dc4e71caba6472ff48ad7d234389b91dadc280
   YogaKit: f782866e155069a2cca2517aafea43200b01fd5a
 
-PODFILE CHECKSUM: 20f7037d54aa3e951a66518d03131ac5563857aa
+PODFILE CHECKSUM: 57257a28933071cf800b998ec6244e943ebfa527
```
Comment (brainbicycle) [inline ios/Podfile.lock:1267]: big updates! woo

File: src/app/Components/StickyTabPage/reanimatedHelpers.ts [hunk 9]
```diff
@@ -6,6 +6,7 @@ import Animated from "react-native-reanimated"
  * given number. Note that the initialization parameter will be ignored
  * on subsequent renders
  * @param init
+ * @deprecated use `useSharedValue` from `react-native-reanimated`
```
Comment (brainbicycle) [inline src/app/Components/StickyTabPage/reanimatedHelpers.ts:9]: 👍

File: src/app/Scenes/Onboarding/OnboardingPersonalization2/OnboardingFollows.tsx [hunk 10]
```diff
@@ -38,6 +38,7 @@ export const OnboardingFollows: React.FC<OnboardingFollowsProps> = ({ kind }) =>
 
   const handleNextButtonPress = () => {
     next()
+    // @ts-expect-error
     navigate("OnboardingPostFollowLoadingScreen")
```
Comment (brainbicycle) [inline src/app/Scenes/Onboarding/OnboardingPersonalization2/OnboardingFollows.tsx:42]: 🤔 why does this error?

File: src/app/Scenes/Search/RecentSearches.tests.tsx [hunk 11]
```diff
@@ -1,111 +1,57 @@
...
+    type: "AUTOSUGGEST_RESULT_TAPPED",
+    props: {
+      displayLabel: name,
+      displayType: "Artist",
+      href: `https://artsy.com/artist/${slug}`,
+      imageUrl: "https://org-name.my-cloud-provider.com/bucket-hash/content-hash.jpg",
+      __typename: "Artist",
+    },
+  })
+)
```
Comment (brainbicycle) [inline src/app/Scenes/Search/RecentSearches.tests.tsx:26]: nice

File: scripts/generate-cities-cache.js [hunk 12]
```diff
@@ -13,13 +13,12 @@ const cities = require("../data/cityDataSortedByDisplayPreference.json")
...
-const FRESHNESS = moment()
-  .add(1, "month")
-  .toDate()
+const FRESHNESS = moment().add(1, "month").toDate()
 
+// @ts-expect-error
 const queryData = queryMap()[QUERY_NAME]
 
-cities.forEach(city => {
+cities.forEach((city) => {
```
[the remark this answers is not in the document]
Comment (brainbicycle) [reply scripts/generate-cities-cache.js:21]: maybe we just exclude them again rather than adding the ignore comments?

**Pull request merged**